src/search/index_manager.h

```diff
@@ -272,6 +273,58 @@ struct IndexManager {
 
     return Status::OK();
   }
-};
 
+  Status FieldValues(engine::Context &ctx, std::string_view index_name, std::string_view input_tag,
+                     const std::string &ns, std::unordered_set<std::string> *ret) {
+    auto iter = index_map.Find(index_name, ns);
+    if (iter == index_map.end()) {
+      return {Status::NotOK, fmt::format("Index '{}' not found in namespace '{}'", index_name, ns)};
+    }
+    const auto &info = iter->second;
+
+    std::string input_tag_str(input_tag);
+    auto field_it = info->fields.find(input_tag_str);
+    if (field_it == info->fields.end()) {
+      return Status::OK();
+    }
+    const auto &[field_name, field_info] = *field_it;
+
+    if (!field_info.metadata || field_info.metadata->type != IndexFieldType::TAG) {
+      return Status::OK();
+    }
+
+    std::unordered_set<std::string> matching_values;
+    util::UniqueIterator index_iter(ctx, ctx.DefaultScanOptions(), ColumnFamilyID::Search);
+
+    auto index_key = SearchKey(ns, index_name, field_name);
+    std::string field_prefix;
+    index_key.PutNamespace(&field_prefix);
+    SearchKey::PutType(&field_prefix, SearchSubkeyType::FIELD);
+    index_key.PutIndex(&field_prefix);
+    PutSizedString(&field_prefix, field_name);
+
+    for (index_iter->Seek(field_prefix); index_iter->Valid(); index_iter->Next()) {
+      auto key = index_iter->key();
+
+      if (!key.starts_with(field_prefix)) {
+        break;
+      }
+
+      Slice key_slice = key;
+      key_slice.remove_prefix(field_prefix.size());
+
+      Slice field_value_slice;
+      if (!GetSizedString(&key_slice, &field_value_slice)) continue;
+
+      matching_values.insert(field_value_slice.ToString());
```

Comment:
Here we can make an optimization: if the tag is still the last one, we can skip it instead of insert it.

Reply:
I have now implemented it, let me know if its fine